homeassistant/helpers/service.py

```diff
@@ -900,8 +895,9 @@ async def entity_service_call(
     )
     assert not pending
 
-    for task in done:
-        task.result()  # pop exception if have
+    response_data: EntityServiceResponse = {}
+    for entity, task in zip(entities, done):
+        response_data[entity.entity_id] = task.result()  # type: ignore[index]
```

Comment:
Whats the typing problem here? It can't match the type of the `response_data` index?

Reply:
It complains that respone_data might be None which does not support indexing

Reply:
Alternative would be `assert response_data is not None`

Reply:
Maybe the `None` should be removed from `EntityServiceResponse` type?

Reply:
`entity_service_call` returns `EntityServiceResponse | None`  so maybe its simplier to propagate `None` explicitly anywhere its needed....

Reply:
Yep that works